tutorial.md

```diff
@@ -0,0 +1,1140 @@
+this really requires more shell-scripting than Haskell knowledge—you can
+also try to seamlessly embed trayer into xmobar by using
+[trayer-padding-icon.sh] and following the advice given in that thread.
+
+In the commands list you, well, define commands.  Commands are the
+pieces that generate the content that is available to display, which
+will later be combined together in the template.  Here, we have defined
+a weather widget, a CPU widget, memory and swap widgets, a date, a
+volume indicator, and of course the data from xmonad via `XMonadLog`.
+
+The `EGPF` in the weather command is a particular station.  Replace both
+(!) occurences of it with your choice of ICAO weather stations.  For a
+list of ICAO codes you can visit the relevant [wikipedia page].  You can
+of course monitor more than one if you like; see xmobar's [weather
+monitor] documentation for further details.
+
+The template then combines them together.  The `alignSep` variable
+controls the alignment of all of the monitors.  Stuff to be
+left-justified goes before the `}` character, things to be centered
+after it, and things to be right justified after `{`.  We have nothing
+centered so there is nothing in between them.
+
+Save the file to `~/.xmobarrc`.  Now you should be able to press `M-q`
+to reload xmonad; this should now display xmobar with your new
+configuration!
+
+It is also possible to completely configure xmobar in Haskell, just like
+xmonad.  If you want to know more about that, you can check out the
+[xmobar.hs] example in the official documentation.  For a more
+complicated example, you can also check out [jao's xmobar.hs] (he's the
+current maintainer of xmobar).
+
+## Changing What XMonad Sends to Xmobar
+
+Now that the xmobar side of the picture looks nice, what about the stuff
+that xmonad sends to xmobar?  It would be nice to visually match these
+two.  Sadly, this is not quite possible with our `xmobarProp` function;
+however, looking at the implementation of the function (or, indeed, the
+top-level documentation of the module!) should give us some ideas for how
+to proceed:
+
+``` haskell
+  xmobarProp config = statusBarProp "xmobar" xmobarPP toggleStrutsKey config
+```
+
+This means that `xmobarProp` just calls the function `statusBarProp`
+with some arguments; crucially for us, notice the `xmobarPP`.  In this
+context "PP" stands for "pretty-printer"—exactly what we want to modify!
+
+_IF YOU ARE ON A VERSION `< 0.17`_: `statusBar` has the exact same
+  relation to `xmobar` as `statusBarProp` has to `xmobarProp`, so just
+  remove the `Prop` and you should be good!
+
+Let's copy the implementation over into our main function:
+
+``` haskell
+  main :: IO ()
+  main = xmonad
+       . ewmhFullscreen
+       . ewmh
+     =<< statusBarProp "xmobar" def toggleStrutsKey myConfig
+    where
+      toggleStrutsKey :: XConfig Layout -> (KeyMask, KeySym)
+      toggleStrutsKey XConfig{ modMask = m } = (m, xK_b)
+```
+
```

Comment:
Maybe it is good to add an example with specifying the xmobar config path, such as
```
statusBarProp "xmobar -x 0 ~/.config/xmobar/xmobarrc0" def toggleStrutsKey myConfig
``` 
Can you provide a multiscreen example here?

Reply:
This sounds like a good idea for the `Further Customizations` section, thanks!  I'd like to wait until monad/xmonad-contrib/pull/463 is finalized though, so we can showcase that functionality straight away.